### PR TITLE
Fix conversion-null, nonnull-compare, dangling-else, address warnings

### PR DESCRIPTION
**Pull request by ozone10**
, add initializers

fix https://github.com/notepad-plus-plus/notepad-plus-plus/issues/12198

### CODE REVIEW

File: PowerEditor/src/MISC/Common/Common.cpp [hunk 1]
```diff
@@ -1285,11 +1285,10 @@ bool isAssoCommandExisting(LPCTSTR FullPathName)
 
 		// check if association exist
 		hres = AssocQueryString(ASSOCF_VERIFY|ASSOCF_INIT_IGNOREUNKNOWN, ASSOCSTR_COMMAND, ext, NULL, buffer, &bufferLen);
-        
+
         isAssoCommandExisting = (hres == S_OK)                  // check if association exist and no error
-			&& (buffer != NULL)                                 // check if buffer is not NULL
```
Comment (ozone10) [inline PowerEditor/src/MISC/Common/Common.cpp:1290]: -Waddress warning: the address of 'buffer' will never be NULL

File: PowerEditor/src/Notepad_plus.cpp [hunk 2]
```diff
@@ -2948,11 +2948,13 @@ bool removeUnwantedTrailingCharFromUrl (TCHAR const *text, int* length)
 				{
 					if (text [j] == closingParenthesis [i])
 						count++;
-					if (text [j] == openingParenthesis [i])
+					if (text[j] == openingParenthesis[i])
```
Comment (ozone10) [inline PowerEditor/src/Notepad_plus.cpp:2951]: -Wdangling-else warning: `else` belong to innermost `if`

File: PowerEditor/src/Notepad_plus.cpp [hunk 3]
```diff
@@ -5937,10 +5939,6 @@ void Notepad_plus::prepareBufferChangedDialog(Buffer * buffer)
 
 void Notepad_plus::notifyBufferChanged(Buffer * buffer, int mask)
 {
-	// To avoid to crash while MS-DOS style is set as default language,
-	// Checking the validity of current instance is necessary.
-	if (!this) return;
```
Comment (ozone10) [inline PowerEditor/src/Notepad_plus.cpp:5942]: `this` should never be null. Some compiler even ignore comparing `this` with `NULL`.
If `this` is null then something bad has happened leading  to undefined behavior.

@donho 
I am getting crash even with this check.

File: PowerEditor/src/ScintillaComponent/Buffer.h [hunk 4]
```diff
@@ -174,9 +174,9 @@ class Buffer final {
 
 	bool isDirty() const { return _isDirty; }
 
-	bool isReadOnly() const { return (_isUserReadOnly || _isFileReadOnly); };
+	bool isReadOnly() const { return (_isUserReadOnly || _isFileReadOnly); }
 
-	bool isUntitled() const { return (_currentStatus == DOC_UNNAMED); }
+	bool isUntitled() const { return ((_currentStatus & DOC_UNNAMED) == DOC_UNNAMED); }
```
Comment (ozone10) [inline PowerEditor/src/ScintillaComponent/Buffer.h:179]: warning from MSVC, use bitwise check

File: PowerEditor/src/ScintillaComponent/UserDefineDialog.cpp [hunk 5]
```diff
@@ -1883,13 +1883,13 @@ intptr_t CALLBACK StylerDlg::dlgProc(HWND hwnd, UINT message, WPARAM wParam, LPA
...
 						const size_t intStrLen = 3;
-						TCHAR intStr[intStrLen];
+						TCHAR intStr[intStrLen] = { '\0' };
 						auto lbTextLen = ::SendDlgItemMessage(hwnd, LOWORD(wParam), CB_GETLBTEXTLEN, i, 0);
 						if (static_cast<size_t>(lbTextLen) > intStrLen - 1)
 							return TRUE;
 
 						::SendDlgItemMessage(hwnd, LOWORD(wParam), CB_GETLBTEXT, i, reinterpret_cast<LPARAM>(intStr));
-                        if ((!intStr) || (!intStr[0]))
+                        if (!intStr[0])
```
Comment (ozone10) [inline PowerEditor/src/ScintillaComponent/UserDefineDialog.cpp:1892]: -Waddress warning: the address of 'intStr' will never be NULL

**Comment by ozone10** [comment]
This should be last PR for gcc warnings from me for now.

-Wunknown-pragmas for `#pragma region` might be ignored by later gcc compiler, IIRC there is patch for it

-Wdelete-non-virtual-dtor
https://ci.appveyor.com/project/donho/notepad-plus-plus/builds/44811960/job/a7qaamkefci41ynt#L489

I will leave this warning for someone else.

**Comment by ArkadiuszMichalski** [comment]
@ozone10 
https://ci.appveyor.com/project/donho/notepad-plus-plus/builds/44818434/job/kp8yuqnp32yldkxx

For 32-bit this one still exist:
```
PowerEditor/src/ScintillaComponent/SmartHighlighter.cpp: In member function 'void SmartHighlighter::highlightViewWithWord(ScintillaEditView*, const generic_string&)':
PowerEditor/src/ScintillaComponent/SmartHighlighter.cpp:72:21: warning: comparison of integer expressions of different signedness: 'long int' and 'long unsigned int' [-Wsign-compare]
  for (; currentLine < lastLine; ++currentLine)
         ~~~~~~~~~~~~^~~~~~~~~~
```

File: PowerEditor/src/ScintillaComponent/Buffer.h [hunk 6]
```diff
@@ -300,15 +300,15 @@ class Buffer final {
...
-	};
+	}
 
 	HANDLE getMonitoringEvent() const { return _eventHandle; };
 
 	void stopMonitoring() { 
 		_isMonitoringOn = false;
 		::SetEvent(_eventHandle);
 		::CloseHandle(_eventHandle);
-	};
```
Comment (donho) [inline PowerEditor/src/ScintillaComponent/Buffer.h:311]: I know it's not necessary to add `;`.
But it's just one of Notepad++ coding convention in *.h

**Review by donho — CHANGES_REQUESTED** [review]
Please restore `;` at the end of function definition in header files.

**Review by donho — APPROVED** [review]

**Pull request closed without merging**